data_structures/Queue/Circular_Queue.dart

```diff
@@ -0,0 +1,78 @@
+// author: kjain1810
+// reference: https://en.wikipedia.org/wiki/Circular_buffer
+const int MAX_SIZE = 10;
+
+class CircularQueue<T> {
+  int start = -1, end = -1;
+  List<T> queue = new List<T>(MAX_SIZE);
+
+  // insert elements into the queue
+  void enque(T element) {
+    if (start == -1) {
+      start = 0;
+      end = 0;
+      queue[0] = element;
+      return;
+    }
+    if (end == MAX_SIZE - 1 && start == 0) {
+      print("The queue is full!!!");
+      return;
+    }
+    if (end == start - 1) {
+      print("The queue is full!!!");
+      return;
+    }
+    end++;
+    end %= MAX_SIZE;
+    queue[end] = element;
+  }
+
+  // remove elements from the queue
+  void deque() {
```

Comment:
Would be great to return removed element. 
This way you could add some test cases